plugins/modules/zos_data_set.py

```diff
@@ -1415,9 +1415,9 @@ def perform_data_set_operations(data_set, state, replace, tmp_hlq, force):
     elif state == "absent":
         changed = data_set.ensure_absent()
     elif state == "cataloged":
-        changed = data_set.ensure_cataloged()
+        changed = data_set.ensure_cataloged(tmp_hlq=tmp_hlq)
     elif state == "uncataloged":
-        changed = data_set.ensure_uncataloged()
+        changed = data_set.ensure_uncataloged(tmp_hlq=tmp_hlq)
     return changed
```

Comment:
how come the rest of these `ensure_*` methods don't get the `tmphlq` passed in? I noticed for example for `ensure_absent`, you added a default value of `tmphlq=None` in the function signature, but what if the user wants to define one, how does that value make it to the function call?

Reply:
Are you only referring to the calls in zos_data_set or also to the functions in the data_set.py util?

The calls in zos_data_set only have `tmp_hlq` when they deal with datasets, the calls that check members don't end up using `mvscmd` and thus don't include the HLQ. For the specific call to `data_set.ensure_absent`, I included the HLQ at first, but then that caused errors while testing, so I removed it.

Reply:
I was referring to:
- `ensure_present`
- `ensure_absent`
- `ensure_cataloged` <-- you've got this one covered
- `ensure_uncataloged` <-- you've got this one covered

For `ensure_absent`, it calls `attempt_catalog_if_necessary_and_delete` which calls `data_set_cataloged` which calls `mvscmd`. 

For `ensure_present`, it calls `attempt_catalog_if_necessary` which calls `data_set_cataloged` which calls `mvscmd`.  

To me, it seems like both `ensure_present` and `ensure_absent` should pass the`tmphlq` all the way down the chain, otherwise, the module option would still be ignored for certain operations. 
Any chance that the errors you got were because the chain of calls wasn't completely updated?

Reply:
I'm not sure what other updates I made to the code, but now the calls don't end up with errors.